### ISSUE TITLE
LUCI_CONTEXT issue with Skia Gold authorization failure - framework_tests in Windows

### ISSUE THREAD

**Issue by keyonghan** [flutter/flutter#63587]
Luci post-submit framework_tests has been consistently hitting error in recent PRs.

```
Failed to load "C:\b\s\w\ir\k\flutter\packages\flutter\test\widgets\absorb_pointer_test.dart":
  Skia Gold authorization failed.
  Luci environments authenticate using the file provided by LUCI_CONTEXT. There may be an error with this file or Gold authentication.
  Debug information for Gold:
  stdout: 
  stderr: Error running command: ''writing to JSON file C:\b\s\w\ir\tmp\t\skia_goldens_8720\packages\flutter\test\widgets\temp\auth_opt.json: writing/serializing to JSON file C:\b\s\w\ir\tmp\t\skia_goldens_8720\packages\flutter\test\widgets\temp\auth_opt.json: Failed to create temporary file for WithWriteFile: pattern contains path separator. At common.go:133 goldauth.go:189 cmd_auth.go:59 command.go:846 command.go:950 main.go:44 proc.go:204 asm_amd64.s:1374''
  
  
  package:flutter_goldens_client/skia_client.dart 171:7                                        SkiaGoldClient.auth
  ===== asynchronous gap ===========================
  package:flutter_goldens/flutter_goldens.dart 252:19                                          FlutterPostSubmitFileComparator.fromDefaultComparator
  package:flutter_goldens/flutter_goldens.dart 32:66                                           main
  ..\..\..\..\tmp\t\flutter_tools.bb5343c7\flutter_test_listener.16d81813\listener.dart 47:18  main.<fn>.<fn>
  ===== asynchronous gap ===========================
  package:test_api                                                                             RemoteListener.start
  ..\..\..\..\tmp\t\flutter_tools.bb5343c7\flutter_test_listener.16d81813\listener.dart 17:25  serializeSuite
  ..\..\..\..\tmp\t\flutter_tools.bb5343c7\flutter_test_listener.16d81813\listener.dart 43:36  main
```

**Comment by keyonghan** [comment]
Marking as P0 as this is blocking the tree.

**Comment by Piinks** [comment]
cc/ @kjlubick @leandrolovisolo
It looks like just the Windows builds are failing: https://ci.chromium.org/p/flutter/builders/prod/Windows

**Comment by Piinks** [comment]
It looks like Go 1.15 contains a breaking change:
> io/ioutil
TempDir and TempFile now reject patterns that contain path separators. That is, calls such as ioutil.TempFile("/tmp", "../base*") will no longer succeed. This prevents unintended directory traversal.
https://golang.org/doc/go1.15

Gold uses a helper library that does not use cross-platform code, which is why we are seeing this failure on the Window builds.
I'm going to pin us to an earlier version of goldctl to fix the build.

**Comment by christopherfujino** [comment]
never would have guessed go was the culprit. there are so many things that can break our build...

**Comment by Piinks** [comment]
This was further obscured by a gap in testing. No framework tests were being executed on windows (Luci or Cirrus) in pre-submit, and only a Windows Luci build was running framework tests in post-submit.
Fix for that is in https://github.com/flutter/flutter/pull/63594

**Comment by Piinks** [comment]
#63593 pins goldctl for Cirrus builds, this will pin for Luci builds: https://flutter-review.googlesource.com/c/recipes/+/5600

**Comment by Piinks** [comment]
Gold has also deployed a fix, will still land pinned versions to prevent future recurrence. Tree should re-open shortly when this build finishes: https://ci.chromium.org/p/flutter/builders/prod/Windows/10341

**Comment by godofredoc** [comment]
Are we ready to close this bug?

**Comment by Piinks** [comment]
Yes, Gold has patched the bug on their side, and #63593 & #65394 will mitigate recurrence. Both changes were held up on the LUCI scheduling issue resolved in https://flutter-review.googlesource.com/c/recipes/+/5540
#63593 should land when the tree turns green, it looks like #63594 needs an update @godofredoc 
I'll close this since the tree is no longer blocked.

**Comment by godofredoc** [comment]
I'll abandon #63594 and create a new one to use the sharded ones.
